lib/pbench/server/indexing_tarballs.py

```diff
@@ -0,0 +1,492 @@
+                        tb_res = 10
+                    except tarfile.TarError as e:
+                        idxctx.logger.error(
+                            "Can't unpack tar ball into {}: {}", ptb.extracted_root, e
+                        )
+                        tb_res = 11
+                    except SigTermException:
+                        idxctx.logger.exception(
+                            "Indexing interrupted by SIGTERM, terminating"
+                        )
+                        break
+                    except Exception as e:
+                        idxctx.logger.exception("Other indexing error: {}", e)
+                        tb_res = 12
+                    else:
+                        beg, end, successes, duplicates, failures, retries = es_res
+                        idxctx.logger.info(
+                            "done indexing (start ts: {}, end ts: {}, duration:"
+                            " {:.2f}s, successes: {:d}, duplicates: {:d},"
+                            " failures: {:d}, retries: {:d})",
+                            tstos(beg),
+                            tstos(end),
+                            end - beg,
+                            successes,
+                            duplicates,
+                            failures,
+                            retries,
+                        )
+                        tb_res = 1 if failures > 0 else 0
+                    try:
+                        ie_len = ie_filepath.stat().st_size
+                    except FileNotFoundError:
+                        # Above operation never made it to actual indexing, ignore.
+                        pass
+                    except SigTermException:
+                        # Re-raise a SIGTERM to avoid it being lumped in with
+                        # general exception handling below.
+                        raise
+                    except Exception:
+                        idxctx.logger.exception(
+                            "Unexpected error handling" " indexing errors file: {}",
+                            ie_filepath,
+                        )
+                    else:
+                        # Success fetching indexing error file size.
+                        if ie_len > len(tb) + 1:
+                            try:
+                                report.post_status(tstos(end), "errors", ie_filepath)
+                            except Exception:
+                                idxctx.logger.exception(
+                                    "Unexpected error issuing"
+                                    " report status with errors: {}",
+                                    ie_filepath,
+                                )
+                    finally:
+                        # Unconditionally remove the indexing errors file.
+                        try:
+                            os.remove(ie_filepath)
+                        except SigTermException:
```

Comment:
Do we need to handle SigTermException here?

Reply:
Doesn't seem like right here

Reply:
@ndokos, see the comment on lines 347 and 348. If there is a better way to word that comment to make this requirement clear, let me know.